### DISCUSSION TITLE
DbApiHook run mutisql can not return last correct

### DISCUSSION THREAD

**Discussion by chuxiangfeng** [apache/airflow#31528] [General]
### Apache Airflow version

Other Airflow 2 version (please specify below)

apache-airflow==2.6.0
apache-airflow-providers-cncf-kubernetes==6.1.0
apache-airflow-providers-common-sql==1.4.0
apache-airflow-providers-dingding==3.1.0
apache-airflow-providers-elasticsearch==4.4.0
apache-airflow-providers-ftp==3.3.1
apache-airflow-providers-http==4.3.0
apache-airflow-providers-imap==3.1.1
apache-airflow-providers-mongo==3.1.1
apache-airflow-providers-mysql==5.0.0
apache-airflow-providers-postgres==5.4.0
apache-airflow-providers-redis==3.1.0
apache-airflow-providers-sqlite==3.3.2

### What happened

pass an List[sqlstr] to DbApiHook run, it can not return last correct.

### What you think should happen instead

return last sql fetch result.
the source code in DbApiHook on line 353 is
```
for sql_statement in sql_list:
      self._run_command(cur, sql_statement, parameters)

      if handler is not None:
          result = handler(cur)
          if return_single_query_results(sql, return_last, split_statements):  //353 line
              _last_result = result
              _last_description = cur.description
          else:
              results.append(result)
              self.descriptions.append(cur.description)
```
i overwrite get_records in sub class with this, change sql to sql_statement,  it make effect.
```
for sql_statement in sql_list:
    self._run_command(cur, sql_statement, parameters)

    if handler is not None:
        result = handler(cur)
        if return_single_query_results(sql_statement, return_last, split_statements):  //353 line
            _last_result = result
            _last_description = cur.description
        else:
            results.append(result)
            self.descriptions.append(cur.description)
```

### How to reproduce

call get_records in DbApiHook.

### Operating System

PRETTY_NAME="Debian GNU/Linux 11 (bullseye)" NAME="Debian GNU/Linux" VERSION_ID="11" VERSION="11 (bullseye)" VERSION_CODENAME=bullseye ID=debian HOME_URL="https://www.debian.org/" SUPPORT_URL="https://www.debian.org/support" BUG_REPORT_URL="https://bugs.debian.org/"

### Deployment

Official Apache Airflow Helm Chart

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
I have no idea what the problem is, and whether there is any issue at all. 

Converting into a discussion. If you want help, please make sure that you spend time on explaining what your problem is and what help you are looing for.  Feel free to be more eloquent in this discussion.

**Reply by chuxiangfeng** [reply]
When I call DbApiHook run and pass multiple sql statements as described in the method, it always returns the execution result of the last sql statement. But it actually returns multiple results, and the others are empty. The following example：
DbApiHook.run['set time zone 'PRC'','select * from aaa'],In theory it should only return` select * from aaa `results，Because return_last is set to true，but In fact， it return (None, results for sql `select * from aaa`) as a tuple，I think there's something wrong with this line of code on line 353，When executing sql through a loop，it should pass loop variable sql_statement as sql to return_single_query_results Instead of sql array passed at the beginning of the method.

**Reply by potiuk** [reply]
Sure . Why don't you create a PR fixing it  and discuss it there?

**Reply by potiuk** [reply]
You seem to know exactly what is wrong and how to fix it
